cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -332,15 +346,65 @@ public boolean accept(String className, URL classUrl, URL classPathUrl) {
     return jarFiles;
   }
 
+  public static File hijackConfFile(File confFile) {
+    if (HIVE_SITE_FILE_PATTERN.matcher(confFile.getAbsolutePath()).matches()) {
+      return hijackHiveConfFile(confFile);
+    } else if (YARN_SITE_FILE_PATTERN.matcher(confFile.getAbsolutePath()).matches()) {
+      return hijackYarnConfFile(confFile);
+    } else {
+      return confFile;
+    }
+  }
+
   /**
-   * Check that the file is a hive-site.xml file, and return a temp copy of it to which are added
-   * necessary options. If it is not a hive-site.xml file, return it as is.
+   * Check that the file is a yarn-site.xml file, and return a temp copy of it to which are added
+   * necessary options. If it is not a yarn-site.xml file, return it as is.
    */
-  public static File hijackHiveConfFile(File confFile) {
-    if (!HIVE_SITE_FILE_PATTERN.matcher(confFile.getAbsolutePath()).matches()) {
-      return confFile;
+  private static File hijackYarnConfFile(File confFile) {
+    Configuration conf = new Configuration(false);
+    try {
+      conf.addResource(confFile.toURI().toURL());
+    } catch (MalformedURLException e) {
+      LOG.error("File {} is malformed.", confFile, e);
+      throw Throwables.propagate(e);
     }
 
+    String yarnAppClassPath = conf.get(YarnConfiguration.YARN_APPLICATION_CLASSPATH,
+                                       Joiner.on(",").join(YarnConfiguration.DEFAULT_YARN_APPLICATION_CLASSPATH));
+
+    // add the pwd/* at the beginning of classpath. so user's jar will take precedence and without this change,
+    // job.jar will be at the begging of the classpath, since job.jar has old guava version classes,
+    // we want to add pwd/* before
+    yarnAppClassPath = "$PWD/*," + yarnAppClassPath;
+
+    conf.set(YarnConfiguration.YARN_APPLICATION_CLASSPATH, yarnAppClassPath);
+
+    File newYarnConfFile = new File(Files.createTempDir(), "yarn-site.xml");
+    FileOutputStream fos;
+    try {
+      fos = new FileOutputStream(newYarnConfFile);
+    } catch (FileNotFoundException e) {
+      LOG.error("Problem creating temporary yarn-site.xml conf file at {}", newYarnConfFile, e);
+      throw Throwables.propagate(e);
+    }
+
+    try {
+      conf.writeXml(fos);
+    } catch (IOException e) {
+      LOG.error("Could not write modified configuration to temporary yarn-site.xml at {}", newYarnConfFile, e);
+      throw Throwables.propagate(e);
+    } finally {
+      Closeables.closeQuietly(fos);
```

Comment:
Shouldn't do close quietly. It there is exception during close, meaning the file content may not be right. Also, I dont't think we need to distinguish file not found vs io exception for the logging purpose since we are logging the exception already. BTW, use the new ARM syntax:

```java
try {
  try (OutputStream os = new FileOutputStream(newYarnConfFile)) {
    conf.writeXml(os);
  }
} catch (IOException e) {
  LOG.error(...);
  throw Throwables.propagate(e);
}
```